### ISSUE TITLE
Feature request: ability to set `Automatically roxygenize when running` options in `*.Rproj` files

### ISSUE THREAD

**Issue by mkearney** [rstudio/rstudio#2900]
When working in Rstudio projects (for writing R packages), I would like to be able to control/customize `Automatically roxygenize when running` via the `*.Rproj` file.

### System details

    RStudio Edition : Desktop
    RStudio Version : Version 1.1.446
    OS Version      : macOS High Sierra Version 10.13.4
    R Version       : R-devel_2018-03-22_r74438

### Steps to reproduce the problem

    1. Create .Rproj R package
    2. Open `*.Rproj` file in text editor
    3. In the `Build` tab (env pane), click `⚙️ More ▾` and then `🛠 Configure Build Tools...`
    4. In `Project Options` pop-up, click `Configure...`
    5. Under `Automatically roxygenize when running:` check the box next to `Build & Reload`
    6. Refresh `*.Rproj` file in text editor and see no change in settings

### Describe the problem in detail

    Changing the settings in the project settings GUI does not translate to changes 
    in the `*.Rproj` file (and vice versa).

### Describe the behavior you expected

    I would like there to be an API for `Automatically roxygenize when running` so I can 
    streamline creation of projects without having to interactively customize settings.

### Closing note

    I'm quite happy with the R project experience regardless of whether this is possible. 
    I just figured I'd mention it because I would certainly appreciate it if it happened!

**Comment by ronblum** [comment]
@mkearney Thank you for the suggestion. However, in step 4, how are you getting to a "Configure..." button? The "Projects Options" pop-up that comes up after step 3 doesn't have it:
<img width="543" alt="build_options" src="https://user-images.githubusercontent.com/31009952/40990786-e89ea278-68bf-11e8-9f39-879d701f9874.png">

**Comment by coatless** [comment]
@ronblum in order to access the "configure" button for "Generate documentation with Roxygen", you _must_ have `roxygen2` installed. I think it would be great if this option could just be present by default and `roxygen2` was added to the dependency manager. 

c.f. menu when `roxygen2` is installed: 

![screen shot 2018-06-09 at 2 06 29 pm](https://user-images.githubusercontent.com/833642/41195174-5acf0bc2-6bee-11e8-92d1-0c3699d1751b.png)

**Comment by salim-b** [comment]
This is a really annoying issue.

![Bildschirmfoto von 2020-10-23 01-29-37](https://user-images.githubusercontent.com/20040931/96940831-4216d100-14c0-11eb-8e9f-10073c4018b1.png)

_Half_ (encircled in green) of the **Roxygen Options** are already covered by the key `PackageRoxygenize` in the `.Rproj` config file:

```dcf
PackageRoxygenize: rd,collate,namespace
```

But it lacks a second key like `PackageRoxygenizeAuto` for the other half (encircled in red).

Would be really cool if the missing pieces could be added.

**Comment by salim-b** [comment]
This ist still an issue with latest stable version 1.4.1103 of RStudio, see [my comment above](https://github.com/rstudio/rstudio/issues/2900#issuecomment-714822483).

**Comment by salim-b** [comment]
This issue still persists in latest stable RStudio 2021.09.2+382.

**Comment by coatless** [comment]
Coming up on this issues 7th year birthday! 🎉 

I think the ask again is twofold:

1. Show "generate with roxygen2" if `roxygen2` is unchecked & just auto install in the background; and
2. Given expanded configuration options is, add a new key for the lower part of project options, e.g. 

```
PackageRoxygenizeAuto: check, build, installandreload
```

Again, for package dev with students this is a meh experience as a result.

**Comment by kevinushey** [comment]
In the latest daily builds of RStudio, we now set all of these by default for new R package projects:

![Image](https://github.com/user-attachments/assets/b75a2d37-08e4-47c0-a806-609640a8dcb4)

@mkearney, do you think this will be sufficient?

**Comment by ronblum** [comment]
Verified in RStudio Desktop 2025.08.0-daily+256 on Windows 11. Made sure that unsetting the values persists.
